[packagers] Broken java-archive launcher in windows

Originally in https://github.com/spring-projects/spring-cli/issues/178 but noticed same issue in jreleaser as well. `spring-cli` is using fatjar while jreleaser uses exploded jar but both ways fail with:

```
Error: Could not find or load main class
Caused by: java.lang.ClassNotFoundException:
```
Problem is in https://github.com/jreleaser/jreleaser/blob/5ba315d18b85e3937c43663fc418e03cf4b13899/core/jreleaser-templates/src/main/resources/META-INF/jreleaser/templates/java-archive/java-archive/bin/launcher.bat.tpl#L57 which is causing extra argument with space.

I took [jreleaser-standalone-early-access-windows-x86_64.zip](https://github.com/jreleaser/jreleaser/releases/download/early-access/jreleaser-standalone-early-access-windows-x86_64.zip) and enabled "echo" in bat file:

```
C:\Users\jannev\Downloads\jreleaser-standalone-early-access-windows-x86_64\jreleaser-standalone-early-access-windows-x86_64\bin>set JAVA_OPTS=" "

C:\Users\jannev\Downloads\jreleaser-standalone-early-access-windows-x86_64\jreleaser-standalone-early-access-windows-x86_64\bin>""C:\Users\jannev\Downloads\jreleaser-standalone-early-access-windows-x86_64\jreleaser-standalone-early-access-windows-x86_64"\bin\java"  " " -cp """C:\Users\jannev\Downloads\jreleaser-standalone-early-access-windows-x86_64\jreleaser-standalone-early-access-windows-x86_64\jars"\*"" org.jreleaser.cli.Main
Error: Could not find or load main class
Caused by: java.lang.ClassNotFoundException:
```

In `spring-cli` it looks like:

```
C:\Users\jannev\Downloads\spring-cli-standalone-0.9.0-SNAPSHOT-windows-x86_64>set JAVA_OPTS=" "

C:\Users\jannev\Downloads\spring-cli-standalone-0.9.0-SNAPSHOT-windows-x86_64>""C:\Users\jannev\Downloads\spring-cli-standalone-0.9.0-SNAPSHOT-windows-x86_64"\bin\java"  " " -cp ""C:\Users\jannev\Downloads\spring-cli-standalone-0.9.0-SNAPSHOT-windows-x86_64\jars"\*" -jar ""C:\Users\jannev\Downloads\spring-cli-standalone-0.9.0-SNAPSHOT-windows-x86_64\jars"\spring-cli-0.9.0-SNAPSHOT.jar"
Error: Could not find or load main class
Caused by: java.lang.ClassNotFoundException:
```

You get same error in java by just doing:
```
$ java " "
Error: Could not find or load main class  
Caused by: java.lang.ClassNotFoundException:  
```

I could try to PR a fix?

--- a/core/jreleaser-templates/src/main/resources/META-INF/jreleaser/templates/java-archive/java-archive/bin/launcher.bat.tpl
+++ b/core/jreleaser-templates/src/main/resources/META-INF/jreleaser/templates/java-archive/java-archive/bin/launcher.bat.tpl
@@ -32,7 +32,7 @@ goto fail
 
 :findJavaFromJavaHome
 set JAVA_HOME=%JAVA_HOME:"=%
-set JAVA_EXE=%JAVA_HOME%/bin/java.exe
+set JAVA_EXE=%JAVA_HOME%\bin\java.exe
 
 if exist "%JAVA_EXE%" goto execute
 
@@ -47,14 +47,16 @@ goto fail
 :execute
 @rem Setup the command line
 
-set JARSDIRS="%APP_HOME%\lib"
+set JARSDIRS=%APP_HOME%\lib
 {{#distributionJavaMainModule}}
-set CLASSPATH="%JARSDIRS%"
+set CLASSPATH=%JARSDIRS%
 {{/distributionJavaMainModule}}
 {{^distributionJavaMainModule}}
-set CLASSPATH="%JARSDIRS%\*"
+set CLASSPATH=%JARSDIRS%\*
 {{/distributionJavaMainModule}}
-set JAVA_OPTS="%JAVA_OPTS% {{distributionJavaOptions}}"
+{{#distributionJavaOptions}}
+set JAVA_OPTS=%JAVA_OPTS% {{.}}
+{{/distributionJavaOptions}}
 
 @rem Execute
 {{#distributionJavaMainModule}}

--- a/core/jreleaser-templates/src/main/resources/META-INF/jreleaser/templates/jlink/jlink/bin/launcher.bat.tpl
+++ b/core/jreleaser-templates/src/main/resources/META-INF/jreleaser/templates/jlink/jlink/bin/launcher.bat.tpl
@@ -48,16 +48,18 @@ for %%i in ("%~dp0..") do set "BASEDIR=%%~fi"
 
 :javaSetup
 
-set JAVA_HOME="%BASEDIR%"
+set JAVA_HOME=%BASEDIR%
 set JAVACMD="%JAVA_HOME%\bin\java"
-set JARSDIRS="%BASEDIR%\jars"
+set JARSDIRS=%BASEDIR%\jars
 {{#distributionJavaMainModule}}
-set CLASSPATH="%JARSDIRS%"
+set CLASSPATH=%JARSDIRS%
 {{/distributionJavaMainModule}}
 {{^distributionJavaMainModule}}
-set CLASSPATH="%JARSDIRS%\*"
+set CLASSPATH=%JARSDIRS%\*
 {{/distributionJavaMainModule}}
-set JAVA_OPTS="%JAVA_OPTS% {{distributionJavaOptions}}"
+{{#distributionJavaOptions}}
+set JAVA_OPTS=%JAVA_OPTS% {{.}}
+{{/distributionJavaOptions}}
 
 @REM Reaching here means variables are defined and arguments have been captured
 :endInit

